File: VB/Dashboard_CustomSchemaProvider/Form1.vb
Imports DevExpress.DashboardWin

Namespace Dashboard_CustomSchemaProvider
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm

		Public Sub New()
			InitializeComponent()
			dashboardDesigner1.CreateRibbon()
			AddHandler dashboardDesigner1.CustomizeDashboardTitle, AddressOf DashboardDesigner1_CustomizeDashboardTitle

			dashboardDesigner1.LoadDashboard("Data\dashboard-with-products.xml")
		End Sub

		Private Sub DashboardDesigner1_CustomizeDashboardTitle(ByVal sender As Object, ByVal e As DevExpress.DashboardWin.CustomizeDashboardTitleEventArgs)
			Dim itemSave As New DashboardToolbarItem(Sub(args) UseCustomDBSchemaProvider()) With {.Caption = "Use Custom DBSchema Provider"}
			e.Items.Clear()
			e.Items.Add(itemSave)
		End Sub

		Private Sub UseCustomDBSchemaProvider()
			dashboardDesigner1.CustomDBSchemaProviderEx = New CustomDBSchemaProvider()
		End Sub
	End Class
End Namespace

File: CS/Dashboard_CustomSchemaProvider/Form1.cs
using DevExpress.DashboardWin;

namespace Dashboard_CustomSchemaProvider
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            dashboardDesigner1.CreateRibbon();
            dashboardDesigner1.CustomizeDashboardTitle += DashboardDesigner1_CustomizeDashboardTitle;

            dashboardDesigner1.LoadDashboard(@"Data\dashboard-with-products.xml");
        }

        private void DashboardDesigner1_CustomizeDashboardTitle(object sender, DevExpress.DashboardWin.CustomizeDashboardTitleEventArgs e)
        {
            DashboardToolbarItem itemSave = new DashboardToolbarItem(
                (args) => UseCustomDBSchemaProvider())
            {
                Caption = "Use Custom DBSchema Provider",
            };
            e.Items.Clear();
            e.Items.Add(itemSave);
        }

        private void UseCustomDBSchemaProvider()
        {
            dashboardDesigner1.CustomDBSchemaProviderEx = new CustomDBSchemaProvider();
        }
    }
}
